targets/psa_switch/psa_switch.cpp

```diff
@@ -308,16 +308,22 @@ PsaSwitch::ingress_thread() {
     input_buffer.pop_back(&packet);
     if (packet == nullptr) break;
 
-    Parser *parser = this->get_parser("ingress_parser");
-    Pipeline *ingress_mau = this->get_pipeline("ingress");
-    phv = packet->get_phv();
-
     port_t ingress_port = packet->get_ingress_port();
     (void) ingress_port;
     BMLOG_DEBUG_PKT(*packet, "Processing packet received on port {}",
                     ingress_port);
 
+    phv = packet->get_phv();
+
+    Parser *parser = this->get_parser("ingress_parser");
     parser->parse(packet.get());
+
+    if (phv->has_field("psa_ingress_output_metadata.drop")) {
```

Comment:
You should be able to skip all of the has_field() checks for fields that were verified that they existed by an add_required_field() call in the PsaSwitch constructor.  If the field did not exist in the BMv2 JSON file, then an error would have occurred then, so you know it is guaranteed to exist when processing each packet.

Reply:
Agreed. Fixed all the instances of this that I saw while making the other changes.